### PR TITLE
Test case automation for ceph-83594544

### PR DESCRIPTION
**Pull request by chaitanyadommeti**
# Description

Added module for test ceph-83594544 - Disallow Group Snapshot Rollback if Memberships Don't Match 
Also, added test details in suite "suites/squid/rbd/tier-2_rbd_group.yaml"

Success log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-J0VI9G/

### CODE REVIEW

File: suites/squid/rbd/tier-2_rbd_group.yaml [hunk 1]
```diff
@@ -164,3 +164,20 @@ tests:
...
+        rep_pool_config:
+          num_pools: 2
+          num_images: 1
+        ec_pool_config:
+          num_pools: 2
+          num_images: 1
+        fio:
+          size: 10M
+      module: test_rbd_disallow_group_snapshot_rollback.py
+      name: Disallow Group Snapshot Rollback if Memberships Don't Match
```
Comment (sunilangadi2) [inline suites/squid/rbd/tier-2_rbd_group.yaml:182]: make it as `Dis-allow`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+        )
+        pool_types = list()
+        if kw.get("client_node"):
+            client = get_node_by_id(kw.get("ceph_cluster"), kw.get("client_node"))
+        else:
+            client = kw.get("ceph_cluster").get_nodes(role="client")[0]
+        rbd_obj = initial_rbd_config(**kw)
+        pool_types = rbd_obj.get("pool_types")
+        if rbd_obj:
+            log.info("Executing test on Replication and EC pool")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:367]: make it as `Replicated`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+        pools = list(multi_pool_config.keys())
+        test_pool = pools[0]
+        test_group = kw.get("config", {}).get("group", "image_group_default")
+        group_spec = f"{test_pool}/{test_group}"
+        image_spec = []
+
+        for pool, pool_config in multi_pool_config.items():
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+            # write some data to that image using fio or rbd bench
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:76]: remove `or rbd bench`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+        # Add images to the group
+        for ispec in image_spec:
+            add_image_group_kw = {
+                "group-spec": group_spec,
+                "image-spec": ispec,
+                "client": client,
+            }
+            rc = add_image_to_group_and_verify(**add_image_group_kw)
+            if rc:
+                log.error(f"Image {ispec} add to group {group_spec} failed.")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:132]: keep it as `adding`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+                return 1
+            else:
+                log.info(f"Image {ispec} add to group {group_spec} succeeded.")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:135]: keep it as `adding`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+        md5_after_snap = []
+        for i in range(0, len(image_spec)):
+            md5 = get_md5sum_rbd_image(
+                image_spec=image_spec[i],
+                rbd=rbd,
+                client=client,
+                file_path="file" + random_string(len=5),
+            )
+            md5_after_snap.append(md5)
+        log.info(f"md5 sums after snap: {md5_after_snap}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:204]: please keep it it is just `sum`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+                return 1
+
+        out, err = rbd.group.image.rm(
+            **{"group-spec": group_spec, "image-spec": image_spec[0]}
+        )
+        if err:
+            log.error(f"Failed to remove {image_spec[0]}from group {group_spec}")
+            return 1
+        else:
+            log.info(f"Removed {image_spec[0]}from group {group_spec}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:219]: space is missing before `from`

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+        md5_before_snap2 = []
+        for i in range(0, len(image_spec)):
+            md5 = get_md5sum_rbd_image(
+                image_spec=image_spec[i],
+                rbd=rbd,
+                client=client,
+                file_path="file" + random_string(len=5),
+            )
+            md5_before_snap2.append(md5)
+        log.info(f"md5 sums before second snap is: {md5_before_snap2}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:270]: `sum` here as well

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+            if md5_before_snap2[i] == md5_after_rollback[i]:
+                log.info(
+                    f"md5 sum of image {image_spec[i]} is same after group snap rollback operation"
+                )
+            else:
+                log.error(
+                    f"md5 sum of image {image_spec[i]} is not same after group snap rollback operation"
+                )
+                return 1
+        if pool_type == "rep_pool_config":
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:340]: we can remove this Is it as it's applicable only to the EC pool

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: suites/squid/rbd/tier-2_rbd_group.yaml (hunk 1 above)
Comment (chaitanyadommeti) [reply suites/squid/rbd/tier-2_rbd_group.yaml:182]: Test name has been maintained the same as that of in polarion. https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83594544. 
Let me know if the test name be changed in polarion as well.

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py (hunk 6 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_disallow_group_snapshot_rollback.py:204]: md5_after_snap is a list of sums of 3 images. Hence sums.

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py (hunk 2 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_disallow_group_snapshot_rollback.py]: Done

File: tests/rbd/test_rbd_disallow_group_snapshot_rollback.py (hunk 3 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_disallow_group_snapshot_rollback.py]: removed

**Review by sunilangadi2 — APPROVED** [review]
LGTM.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**